code/controllers/subsystem/sounds.dm

```diff
@@ -0,0 +1,87 @@
+#define DATUMLESS "NO_DATUM"
+
+SUBSYSTEM_DEF(sounds)
+	name = "Sounds"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_SOUNDS
+	var/static/using_channels_max = CHANNEL_HIGHEST_AVAILABLE		//BYOND max channels
+
+	// Hey uh these two needs to be initialized fast because the whole "things get deleted before init" thing.
+	/// Assoc list, "[channel]" = either the datum using it or TRUE for an unsafe-reserved (datumless reservation) channel
+	var/list/using_channels = list()
+	/// Assoc list datum = list(channel1, channel2, ...) for what channels something reserved.
+	var/list/using_channels_by_datum = list()
+	/// List of all available channels with associations set to TRUE for fast lookups/allocation.
+	var/list/available_channels
+
+/datum/controller/subsystem/sounds/Initialize()
+	setup_available_channels()
+	return ..()
+
+/datum/controller/subsystem/sounds/proc/setup_available_channels()
+	available_channels = list()
+	for(var/i in 1 to using_channels_max)
```

Comment:
Any reason not to use the `CHANNEL_HIGHEST_AVAILABLE` define instead of a static with the same value?

Reply:
ehh mostly just for varediting if i or someone for whatever reason wants to privately reserve upper channels for their own ends mid-round